src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs

```diff
@@ -67,28 +69,47 @@ public async ValueTask ReportDiagnosticsAsync(DotNetWatchContext context, IEnume
             }
         }
 
-        private async ValueTask<bool> VerifyDeltaApplied(DotNetWatchContext context, CancellationToken cancellationToken)
+        private async Task<bool> VerifyDeltaApplied(DotNetWatchContext context, CancellationToken cancellationToken)
         {
             var _receiveBuffer = new byte[1];
             try 
             {
-                var result = await context.BrowserRefreshServer.ReceiveAsync(_receiveBuffer, cancellationToken);
+                ValueWebSocketReceiveResult? result;
+                // Sometimes we can received a Close message from the WebSocket before the
```

Comment:
If you receive a close, then any call to ReceiveAsync afterwards should throw. What was the issue here that's trying to be resolved?

Reply:
Before the browser had sent the acknowledgement byte, we were getting a `Close` message from the browser which was causing this codepath to prematurely return true.

To workaround this, we added do-while to make sure we received the acknowledgement byte from the browser.

Reply:
>we were getting a Close message from the browser

>To workaround this, we added do-while to make sure we received the acknowledgement byte from the browser.

Right, this doesn't make sense, how are you receiving more data after getting a close message?

Reply:
`ReceiveAsync` reads from multiple client sockets so I believe it was receiving the `Close` message from only one of the clients (e.g. closed tab or something) but not the active one.

Reply:
Makes sense now, didn't realize that wasn't an actual websocket. Would it make more sense to change BrowserRefreshServer to not return close messages instead of having this loop?

Reply:
That might be a cleaner option. We already filter out closed clients, we could do something similar with messages too

Reply:
I think we can do that as a follow-up in main to target preview5. Thoughts?